### PR TITLE
Bump in-toto dependency.

### PR DESCRIPTION
**Pull request by mattmoor**
I noticed that there were some breaking changes in the in-toto version pulled in compared with the version in sigstore.  This updates things to HEAD.

### CODE REVIEW

**Comment by mattmoor** [comment]
cc @priyawadhwa @dlorenc

**Comment by mattmoor** [comment]
Hmm...
```
tekton-chains-controller-7f4d9cc466-4mrdl   0/1   CrashLoopBackOff   5     5m10s
```

**Comment by mattmoor** [comment]
Hello old friend:
```
kubectl logs -n tekton-chains      tekton-chains-controller-5bcfbf8484-rmk46
/ko-app/controller flag redefined: log_dir
panic: /ko-app/controller flag redefined: log_dir

```

**Comment by mattmoor** [comment]
Looks like chains is depending on the `cli` package, which pulls in `glog` (conflicts with K8s `klog`).

The two things I found that need to move out to make a `depcheck_test` pass for chains are:
 * `cli.LoadECDSAPrivateKey`
 * `cli.DssePayloadType`
 
I will look into splitting these out in the morning.

**Comment by mattmoor** [comment]
Ok, I am optimistic that this will pass, but I had to tweak an e2e test to reflect a change pulled in via the updated deps.

File: test/testdata/intoto/task-output-image.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "_type": "https://in-toto.io/Statement/v0.1",
-    "predicateType": "https://in-toto.io/Provenance/v0.1",
+    "predicateType": "https://slsa.dev/provenance/v0.1",
```
Comment (mattmoor) [inline test/testdata/intoto/task-output-image.json:3]: @priyawadhwa LMK if this is expected, but this seems to be what the provenance type changed to in `in-toto` 🤷

Comment (dlorenc) [reply test/testdata/intoto/task-output-image.json:3]: this is expected

**Comment by mattmoor** [comment]
Merge conflict, fixing it now

**Comment by mattmoor** [comment]
Rebased on main and updated deps, but I'll need a fresh `/lgtm`

**Pull request merged**